Magento2/Sniffs/Legacy/ClassReferencesInConfigurationFilesSniff.php

```diff
@@ -0,0 +1,296 @@
+     *
+     * @param File $phpcsFile
+     * @param array $elements
+     */
+    private function assertNonFactoryNameTab(File $phpcsFile, array $elements)
+    {
+        foreach ($elements as $element) {
+            if (preg_match('/\//', $element['value']) !== false) {
+                $phpcsFile->addError(
+                    self::ERROR_MESSAGE_CONFIG,
+                    $element['lineNumber'],
+                    self::ERROR_CODE_CONFIG,
+                );
+            }
+        }
+    }
+
+    /**
+     * Format the incoming XML to avoid tags split into several lines.
+     *
+     * @param File $phpcsFile
+     * @return false|string
+     */
+    private function getFormattedXML(File $phpcsFile)
+    {
+        $doc = new DomDocument('1.0');
+        $doc->formatOutput = true;
+        $doc->loadXML($phpcsFile->getTokensAsString(0, 999999));
+        return $doc->saveXML();
+    }
+
+    /**
+     * Parse an XML for references to PHP class names in selected tags or attributes
+     *
+     * @param SimpleXMLElement $xml
+     * @return array
+     */
+    private function collectClassesInConfig(SimpleXMLElement $xml): array
+    {
+        $classes = $this->getValuesFromXmlTagContent(
+            $xml,
+            '
+            /config//resource_adapter | /config/*[not(name()="sections")]//class[not(ancestor::observers)]
+                | //model[not(parent::connection)] | //backend_model | //source_model | //price_model
+                | //model_token | //writer_model | //clone_model | //frontend_model | //working_model
+                | //admin_renderer | //renderer',
+        );
+        $classes = array_merge(
+            $classes,
+            $this->getValuesFromXmlTagAttribute(
+                $xml,
+                '//@backend_model',
+                'backend_model'
+            )
+        );
+        $classes = array_merge(
+            $classes,
+            $this->getValuesFromXmlTagAttribute(
+                $xml,
+                '/config//preference',
+                'type'
+            )
+        );
+        $classes = array_merge(
+            $classes,
+            $this->getValuesFromXmlTagName(
+                $xml,
+                '/logging/*/expected_models/* | /logging/*/actions/*/expected_models/*',
+            )
+        );
+
+        $classes = array_map(
+            function (array $extendedNode) {
+                $extendedNode['value'] = explode('::', trim($extendedNode['value']))[0];
+                return $extendedNode;
+            },
+            $classes
+        );
+
+        return $classes;
+    }
+
+    /**
+     * Parse an XML for references to PHP class names in selected tags or attributes
+     *
+     * @param SimpleXMLElement $xml
+     * @return array
+     */
+    private function collectClassesInLayout(SimpleXMLElement $xml): array
+    {
+        $classes = $this->getValuesFromXmlTagAttribute(
+            $xml,
+            '/layout//@helper',
```

Comment:
I think these paths related to layouts are outdated, can you check @sivaschenko ?